shipper.py

```diff
@@ -226,49 +227,108 @@ def shipment_post_new():
-                            # Add metadata that are in compendium in db:
-                            if 'metadata' in current_compendium:
-                                if 'zenodo' in current_compendium['metadata']:
-                                    md = current_compendium['metadata']['zenodo']
-                                    zenodo_add_metadata(env_repository_zenodo_host, data['deposition_id'], md,
-                                                        env_repository_zenodo_token)
-                        elif data['recipient'] == 'eudat':
-                            data['deposition_id'] = eudat_create_depot(env_repository_eudat_host, env_repository_eudat_token)
-                            data['deposition_url'] = ''.join((env_repository_eudat_host.replace('api', 'records/'), data['deposition_id']))
-                            eudat_add_zip_to_depot(env_repository_eudat_host, data['deposition_id'], file_name, compendium_files, env_repository_eudat_token)
-                            # Add metadata that are in compendium in db:
-                            if 'metadata' in current_compendium:
-                                if 'eudat' in current_compendium['metadata']:
-                                    md = current_compendium['metadata']['eudat']
-                                    eudat_update_md(env_repository_eudat_host, data['deposition_id'], md, env_repository_eudat_token)
+                    # Aquire path to files via env var and id:
+                    compendium_files = os.path.normpath(os.path.join(env_compendium_files, data['compendium_id']))
+                    # Determine state of that compendium: Is is a bag or not, zipped, valid, etc:
+                    compendium_state = files_scan_path(compendium_files)
+                    if not compendium_state == 0:
+                        # Case path does not exist:
+                        if compendium_state == 1:
+                            # Case: Is a bagit bag:
+                            logging.getLogger('bagit').setLevel(logging.CRITICAL)
+                            try:
+                                bag = bagit.Bag(compendium_files)
+                                bag.validate()
+                                status_note(''.join(('Valid bagit bag at <', str(data['compendium_id']), '>')))
+                            except bagit.BagValidationError as e:
+                                status_note(''.join(('! Invalid bagit bag at <', str(data['compendium_id']), '>')))
+                                details = []
+                                for d in e.details:
+                                    details.append(str(d))
+                                    status_note(str(d))
+                                # Exit point for invalid not to be repaired bags
+                                if not data['update_packaging'] in ['true', 'True', 't', '1']:
```

Comment:
Assiduity points for covering different cases. What about `TRUE` ?

I suggest to use [strtobool](https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool) and some "to lower case" function beforehand.

Reply:
somehow this stayed in the code after I notice the form data could not transport a boolean True.
Will fix.
_Assiduity_ not in my active vocabulary. I guess thats assiduity points for you...

Reply:
Or for online translation services... :see_no_evil:

Reply:
`strtobool` not built in. dont want to import because of "keep imports at min policy". will add a helper func

Reply:
But the library is already there, right? I suggest to rely on a function that is used by a lot of other people, instead of re-doing, but it's your call.